src/middleware/nx_verified_telemetry.c

```diff
@@ -15,7 +15,7 @@ static const CHAR device_status_property[]             = "deviceStatus";
 static const CHAR temp_response_description_success[] = "success";
 static const CHAR temp_response_description_failed[]  = "failed";
 
-static CHAR scratch_buffer[200];
+static CHAR scratch_buffer[3500];
```

Comment:
>3500 [](http://example.com/codeflow?start=27&length=4)

im assuming this is related to the DMA read buffer? How is 3500 calculated? Is this enough?

Reply:
yes this is enough
It is computed based on this struct:

typedef struct VT_CURRENTSENSE_RAW_SIGNATURES_READER_STRUCT
{
    VT_CURRENTSENSE_RAW_SIGNATURE_BUFFER raw_signatures_shared_buffer[VT_CS_MAX_SIGNATURES];
    VT_UINT num_raw_signatures;
    VT_FLOAT adc_read_buffer[VT_CS_SAMPLE_LENGTH];
    VT_FLOAT adc_read_sampling_frequency;
    VT_BOOL raw_signature_ongoing_collection;
} VT_CURRENTSENSE_RAW_SIGNATURES_READER;

in the file https://github.com/Azure/Verified-Telemetry/blob/feature/currentsense_update_v2/inc/core/currentsense/vt_cs_api.h